__tests__/pet.test.js

```diff
@@ -0,0 +1,149 @@
+	});
+});
+
+describe('walk', () => {
+	it('increases fitness by 4', () => {
+		pet.fitness = 5;
+		pet.walk();
+		expect(pet.fitness).toEqual(9);
+	});
+	it('stops fitness going over 10', () => {
+		pet.fitness = 9;
+		pet.walk();
+		expect(pet.fitness).toEqual(10);
+	});
+	it('throws an error if the pet is not alive', () => {
+		pet.age = 30;
+		expect(() => pet.walk()).toThrow('Your pet is no longer alive :(');
+	});
+});
+
+describe('feed', () => {
+	it('decreases pet hunger level by 3', () => {
+		pet.hunger = 5;
+		pet.feed();
+		expect(pet.hunger).toEqual(2);
+	});
+	it('stops hunger going less than 0', () => {
+		pet.hunger = 1;
+		pet.feed();
+		expect(pet.hunger).toEqual(0);
+	});
+	it('throws an error if the pet is not alive', () => {
+		pet.age = 30;
+		expect(() => pet.feed()).toThrow('Your pet is no longer alive :(');
+	});
+});
+
+describe('checkUp', () => {
+	it('return "I need a walk" if fitness less than 4 and hunger less than 5', () => {
+		pet.hunger = 1;
+		pet.fitness = 1;
+		expect(pet.checkUp()).toEqual('I need a walk');
+	});
+	it('return "I am hungry" if hunger greater than 4 and fitness greater than 3', () => {
+		pet.hunger = 5;
+		pet.fitness = 9;
+		expect(pet.checkUp()).toEqual('I am hungry');
+	});
+	it('return "I am hungry AND I need a walk" if hunger greater than 4 and fitness less than 4', () => {
+		pet.hunger = 9;
+		pet.fitness = 1;
+		expect(pet.checkUp()).toEqual('I am hungry AND I need a walk');
+	});
+
+	it('return "I feel great!" if hunger less than 5 and fitness greater than 3', () => {
+		expect(pet.checkUp()).toEqual('I feel great!');
+	});
+	it('returns message if the pet is not alive', () => {
+		pet.age = 30;
+		expect(pet.checkUp()).toEqual('Your pet is no longer alive :(');
+	});
+});
+
+describe('isAlive', () => {
+	it('returns false if fitness < 1', () => {
+		pet.fitness = 0;
+		expect(pet.isAlive).toEqual(false);
+	});
+	it('returns true if fitness > 0', () => {
+		expect(pet.isAlive).toEqual(true);
+	});
+	it('returns false if hunger > 9', () => {
+		pet.hunger = 10;
+		expect(pet.isAlive).toEqual(false);
+	});
+	it('returns true if hunger < 10', () => {
+		expect(pet.isAlive).toEqual(true);
+	});
+	it('returns false if age > 29', () => {
+		pet.age = 30;
+		expect(pet.isAlive).toEqual(false);
+	});
+	it('returns true if age < 30', () => {
+		expect(pet.isAlive).toEqual(true);
+	});
+});
+
+describe('adopt a child', () => {
+	it('children property is type object', () => {
+		let child = new Pet('Amelia');
+		pet.adoptChild(child);
+		expect(typeof pet.children).toEqual('object');
+	});
+	it('check first element of array equals child', () => {
+		let child = new Pet('Amelia');
+		pet.adoptChild(child);
+		expect(pet.children[0].name).toEqual('Amelia');
+	});
+});
+describe('have a child', () => {
+	it('children property is type object', () => {
+		pet.haveaChild('Amelia');
+		expect(typeof pet.children).toEqual('object');
+	});
+	it('check first element of array equals Amelia', () => {
```

Comment:
As above regarding this actually testing the name of the first Pet in the array is correct